web/ui/react-app/src/utils/index.ts

```diff
@@ -210,3 +211,11 @@ export const callAll = (...fns: Array<(...args: any) => void>) => (...args: any)
   // eslint-disable-next-line prefer-spread
   fns.filter(Boolean).forEach(fn => fn.apply(null, args));
 };
+
+export const replaceWith = (str: string, needle: string, replace: string) => {
+  return str.replace(new RegExp(needle, 'g'), `${replace}`);
+};
+
+export const isAlertinRule = (rule: Rule): rule is AlertingRule => {
```

Comment:
I think it's not worth having a function just for this comparison, but if you prefer abstracting away the raw strings, we could add a string enum type for the rule types?

Reply:
Oh my idea was simply to have an enum like:

```ts
export enum RuleType {
  recording = 'recording';
  alerting = 'alerting';
}
```

...and then use that for the field type.

Just to not rely on direct string literals everywhere, although TS still checks in either case whether a rule can possibly have a given type (literal or not).

The first type guard example you linked (with `isFish`) checks for the existence of a field, not the value... which I find a bit more indirect than directly checking the type. I think I would use that pattern if I had multiple things that could swim / fly / etc., and I didn't really care what concrete type they are, just that they can do a certain behavior (like with interfaces in Go). If I have two concrete types that I want to render quite differently, I think I prefer what we currently do (comparison on the `type` field). I would like the later example with `typeof` type guards (https://www.typescriptlang.org/docs/handbook/advanced-types.html#typeof-type-guards), but those string comparisons only seem to work for [number, string, boolean, symbol] types.